packages/frontend/src/pages/Login.tsx

```diff
@@ -88,6 +77,17 @@ const Login: FC = () => {
         return <Redirect to={{ pathname: '/' }} />;
     }
 
+    const isDemo = health.data?.mode === LightdashMode.DEMO;
+    if (health.isLoading || isDemo) {
```

Comment:
Can you explain the conditional logic in these two? I'm very confused 😂

Reply:
Yeah, is a bit confusing 🤔  
Basically, we want to show a loading page when the health data is loading or when it's the demo app. 
The check inside is so we only auto login once. It checks if it's the demo app and the login request is not loading. If that request is loading it means we already triggered the mutation once.

Reply:
refactoring this into a hook like before